From a441f9f65bec741385ad5f3ebfc60dfe666c0a77 Mon Sep 17 00:00:00 2001
From: Jonathan Lebon <jonathan@jlebon.com>
Date: Fri, 3 Dec 2021 17:25:33 -0500
Subject: [PATCH 6/8] internal/exec/stages: formalize concept of "no-op"
 configs

Some of the stages had this upfront check for whether we should just
early exit before doing anything. Formalize this a bit by creating an
explicit `isNoOp` function and put those checks in there.
---
 internal/exec/stages/disks/disks.go | 12 ++++++++----
 internal/exec/stages/kargs/kargs.go |  8 ++++++--
 2 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/internal/exec/stages/disks/disks.go b/internal/exec/stages/disks/disks.go
index 3908a762..920490cc 100644
--- a/internal/exec/stages/disks/disks.go
+++ b/internal/exec/stages/disks/disks.go
@@ -65,14 +65,18 @@ func (stage) Name() string {
 	return name
 }
 
+func isNoOp(config types.Config) bool {
+	return len(config.Storage.Disks) == 0 &&
+		len(config.Storage.Raid) == 0 &&
+		len(config.Storage.Filesystems) == 0 &&
+		len(config.Storage.Luks) == 0
+}
+
 func (s stage) Run(config types.Config) error {
 	// Interacting with disks/partitions/raids/filesystems in general can cause
 	// udev races. If we do not need to  do anything, we also do not need to
 	// do the udevadm settle and can just return here.
-	if len(config.Storage.Disks) == 0 &&
-		len(config.Storage.Raid) == 0 &&
-		len(config.Storage.Filesystems) == 0 &&
-		len(config.Storage.Luks) == 0 {
+	if isNoOp(config) {
 		return nil
 	}
 
diff --git a/internal/exec/stages/kargs/kargs.go b/internal/exec/stages/kargs/kargs.go
index 24bd047c..f3ae05c5 100644
--- a/internal/exec/stages/kargs/kargs.go
+++ b/internal/exec/stages/kargs/kargs.go
@@ -60,9 +60,13 @@ func (stage) Name() string {
 	return name
 }
 
+func isNoOp(config types.Config) bool {
+	return len(config.KernelArguments.ShouldExist) == 0 &&
+		len(config.KernelArguments.ShouldNotExist) == 0
+}
+
 func (s stage) Run(config types.Config) error {
-	if len(config.KernelArguments.ShouldExist) == 0 &&
-		len(config.KernelArguments.ShouldNotExist) == 0 {
+	if isNoOp(config) {
 		return nil
 	}
 
-- 
2.35.1

